src/unix/udp.c

```diff
@@ -459,9 +514,13 @@ int uv__udp_try_send(uv_udp_t* handle,
   if (handle->send_queue_count != 0)
     return UV_EAGAIN;
 
-  err = uv__udp_maybe_deferred_bind(handle, addr->sa_family, 0);
-  if (err)
-    return err;
+  if (addr) {
+    err = uv__udp_maybe_deferred_bind(handle, addr->sa_family, 0);
+    if (err)
+      return err;
+  } else {
+    assert(handle->flags & UV__UDP_CONNECTED);
```

Comment:
Does this need to be an assertion, or could we just return an error?

Reply:
I added the assertion as this should never happen. See: https://github.com/santigimeno/libuv/blob/94adb2020490b9774f7b1c12bf4bf7745dd2be72/src/uv-common.c#L338-L339